runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -2943,13 +2944,230 @@ J9::Z::TreeEvaluator::anewArrayEvaluator(TR::Node * node, TR::CodeGenerator * cg
 TR::Register *
 J9::Z::TreeEvaluator::multianewArrayEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    {
+   static char *useDirectHelperCall = feGetEnv("TR_MultiANewArrayEvaluatorUseDirectCall");
+   TR::Compilation *comp = cg->comp();
+   TR_J9VMBase *fej9 = static_cast<TR_J9VMBase *>(comp->fe());
+   TR::Register *targetReg = cg->allocateRegister();
+
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Node *thirdChild = node->getThirdChild();
+
+   if (useDirectHelperCall || !secondChild->getOpCode().isLoadConst() || secondChild->getInt() != 2)
+      {
+      traceMsg(cg->comp(), "multianewArrayEvaluator - More than 2 dimensions, so we can't handle this scenario in JITT'ed code\n");
+      TR::ILOpCodes opCode = node->getOpCodeValue();
+      TR::Node::recreate(node, TR::acall);
+      TR::Register *helperTargetReg = TR::TreeEvaluator::performCall(node, false, cg);
+      TR::Node::recreate(node, opCode);
+      return helperTargetReg;
+      }
```

Comment:
I'm fairly confident we don't need this code. This is already covered in the lowering phase:
https://github.com/eclipse/openj9/blob/8c76311cdcccaa6def680d362df418eca6ae9325/runtime/compiler/env/VMJ9.cpp#L3394-L3449

I don't think we can ever enter this `if` block. IN fact it makes sense for this check to be performed in the lowering phase as such code doesn't need to be replicated across different codegens.

Reply:
I think you're right. It looks like we do a check here to see if there are more than 2 dimensions. And in such a case we transform this into a `TR::acall` node. Going to just do a couple more sanity checks and then will remove the code in the evaluator.